test/ServerSubscriptionTest.php

```diff
@@ -48,30 +52,30 @@ function tearDown()
     function testBasic() {
 
         $subscriptionData = new Subscription();
-        $subscriptionData->accountId =  $this->account->accountId;
-        $subscriptionData->productName = "Sports";
-        $subscriptionData->productCategory = "BASE";
-        $subscriptionData->billingPeriod = "MONTHLY";
-        $subscriptionData->priceList = "DEFAULT";
-        $subscriptionData->externalKey = $this->externalBundleId;
+        $subscriptionData->setAccountId($this->account->getAccountId());
+        $subscriptionData->setProductName('Sports');
+        $subscriptionData->setProductCategory('BASE');
+        $subscriptionData->setBillingPeriod('MONTHLY');
+        $subscriptionData->setPriceList('DEFAULT');
+        $subscriptionData->setExternalKey($this->externalBundleId);
 
         $subscription = $subscriptionData->create($this->user, $this->reason, $this->comment, $this->tenant->getTenantHeaders());
-        $this->assertEquals($subscription->accountId, $subscriptionData->accountId);
-        $this->assertEquals($subscription->productName, $subscriptionData->productName);
-        $this->assertEquals($subscription->productCategory, $subscriptionData->productCategory);
-        $this->assertEquals($subscription->billingPeriod, $subscriptionData->billingPeriod);
-        $this->assertEquals($subscription->externalKey, $subscriptionData->externalKey);
+        $this->assertEquals($subscription->getAccountId(), $subscriptionData->getAccountId());
+        $this->assertEquals($subscription->getProductName(), $subscriptionData->getProductName());
+        $this->assertEquals($subscription->getProductCategory(), $subscriptionData->getProductCategory());
+        $this->assertEquals($subscription->getBillingPeriod(), $subscriptionData->getBillingPeriod());
+        $this->assertEquals($subscription->getExternalKey(), $subscriptionData->getExternalKey());
 
         # Move by a few days -- still in trial -- and change product
         $this->clock->addDays(3, $this->tenant->getTenantHeaders());
-        $subscription->productName = 'Super';
+        $subscription->setProductName('Super');
         $subscriptionRes = $subscription->changePlan(null, null, null, $this->user, $this->reason, $this->comment, $this->tenant->getTenantHeaders());
-        $this->assertEquals($subscriptionRes->productName, 'Super');
+        $this->assertEquals($subscriptionRes->getProductName(), 'Super');
```

Comment:
I checked it again, according to a request with CURL the subscription has the `sports-monthly-1` plan (the same as we get with the PHP client). The clock gets correctly set forward according to a CURL request to `http://localhost:8080/1.0/kb/test/clock`.

Could it be that just the `productName` isn't being changed correctly? Can you reproduce this somehow?

This is the contents from the `subscription_events` table now: 

```
MariaDB [killbill]> select * from subscription_events\G;
*************************** 1. row ***************************
              record_id: 1
                     id: 589e8515-a990-4834-b5f9-e2bd370878d2
             event_type: API_USER
              user_type: CREATE
         effective_date: 2017-04-06 09:31:59
        subscription_id: 09e6e6c7-09f2-4c58-87a9-54a491f0d8f8
              plan_name: sports-monthly
             phase_name: sports-monthly-trial
        price_list_name: DEFAULT
billing_cycle_day_local: 0
              is_active: 1
             created_by: phpTester
           created_date: 2017-04-06 09:31:59
             updated_by: phpTester
           updated_date: 2017-04-06 09:31:59
      account_record_id: 1
       tenant_record_id: 1
*************************** 2. row ***************************
              record_id: 2
                     id: 5ece97b5-dbbf-4f83-b802-d70babdff4d6
             event_type: PHASE
              user_type: NULL
         effective_date: 2017-05-06 09:31:59
        subscription_id: 09e6e6c7-09f2-4c58-87a9-54a491f0d8f8
              plan_name: NULL
             phase_name: sports-monthly-evergreen
        price_list_name: NULL
billing_cycle_day_local: 0
              is_active: 0
             created_by: phpTester
           created_date: 2017-04-06 09:31:59
             updated_by: phpTester
           updated_date: 2017-04-09 09:32:05
      account_record_id: 1
       tenant_record_id: 1
*************************** 3. row ***************************
              record_id: 3
                     id: f546f56c-de60-4498-b6dc-edfe5b12153a
             event_type: API_USER
              user_type: CHANGE
         effective_date: 2017-04-09 09:32:05
        subscription_id: 09e6e6c7-09f2-4c58-87a9-54a491f0d8f8
              plan_name: sports-monthly-1
             phase_name: sports-monthly-1-trial
        price_list_name: DEFAULT
billing_cycle_day_local: 0
              is_active: 1
             created_by: phpTester
           created_date: 2017-04-09 09:32:05
             updated_by: phpTester
           updated_date: 2017-04-09 09:32:05
      account_record_id: 1
       tenant_record_id: 1
*************************** 4. row ***************************
              record_id: 4
                     id: ef07c457-b0e1-4567-8f4a-5311848bfa60
             event_type: PHASE
              user_type: NULL
         effective_date: 2017-05-06 09:31:59
        subscription_id: 09e6e6c7-09f2-4c58-87a9-54a491f0d8f8
              plan_name: NULL
             phase_name: sports-monthly-1-evergreen
        price_list_name: NULL
billing_cycle_day_local: 0
              is_active: 1
             created_by: phpTester
           created_date: 2017-04-09 09:32:05
             updated_by: phpTester
           updated_date: 2017-04-09 09:32:05
      account_record_id: 1
       tenant_record_id: 1
4 rows in set (0.01 sec)
```